gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/driver/ClientAuthenticationIntegrateTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.tinkerpop.gremlin.driver;
+
+import org.apache.tinkerpop.gremlin.server.AbstractGremlinServerIntegrationTest;
+import org.apache.tinkerpop.gremlin.server.Settings;
+import org.apache.tinkerpop.gremlin.server.TestClientFactory;
+import org.apache.tinkerpop.gremlin.server.auth.AuthenticatedUser;
+import org.apache.tinkerpop.gremlin.server.auth.AuthenticationException;
+import org.apache.tinkerpop.gremlin.server.auth.SimpleAuthenticator;
+import org.junit.Test;
+
+import java.util.Map;
+import java.util.concurrent.Future;
+
+import static org.junit.Assert.assertEquals;
+
+public class ClientAuthenticationIntegrateTest extends AbstractGremlinServerIntegrationTest {
+
+    public static class Authenticator extends SimpleAuthenticator {
+
+        @Override
+        public void setup(Map<String, Object> config) {
+        }
+
+        @Override
+        public AuthenticatedUser authenticate(Map<String, String> credentials) throws AuthenticationException {
+            try {
+                Thread.sleep(1000);
+                return new AuthenticatedUser(credentials.get("username"));
+            } catch (Exception ex) {
+                throw new AuthenticationException(ex);
+            }
+        }
+    }
+
+    /**
+     * Configure specific Gremlin Server settings for specific tests.
+     */
+    @Override
+    public Settings overrideSettings(final Settings settings) {
+        final Settings.AuthenticationSettings authSettings = new Settings.AuthenticationSettings();
+        authSettings.authenticator = Authenticator.class.getName();
+        authSettings.enableAuditLog = true;
+        return settings;
+    }
+
+    @Test
+    public void shouldBeThreadSafeWithAuthentication() throws Exception {
```

Comment:
This test passes on the tp33 branch - if it was testing the problem described in the JIRA issue I would have expected it to fail. Thoughts?

I would instead use this test:

https://issues.apache.org/jira/browse/TINKERPOP-2132?focusedCommentId=16743980&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16743980

According to my comments and to the best of my knowledge it was failing as described in the issue prior to my change in my branch. In that branch I added it to: `GremlinServerAuthIntegrateTest.java` (please just use that rather than creating a new "authentication" test class.

Reply:
Actually, I would add all the tests that I had in my branch - you can see them here: 

https://github.com/apache/tinkerpop/compare/tp33...TINKERPOP-2132

If you can get all those tests to pass then I think you've solved the problem.